src/EventQueue.js

```diff
@@ -1,22 +1,61 @@
 class EventQueue {
   constructor() {
+    this.notFound = true;
     this.queue = [];
+    this.notFoundQueue = [];
   }
 
   suscribe(callback) {
+    const matched = callback();
+
+    if (matched) {
+      if (this.notFound) {
+        this.hideNotFoundRoutes();
+      }
+      this.notFound = false;
+    }
     this.queue.push(callback);
   }
 
+  suscribeNotFound(callback) {
+    callback(!this.notFound);
+    this.notFoundQueue.push(callback);
+  }
+
   unsuscribe(callback) {
-    const index = this.queue.indexOf(callback);
+    this._unsuscribe(callback, this.queue);
+    this._unsuscribe(callback, this.notFoundQueue);
+  }
+
+  _unsuscribe(callback, queue) {
+    const index = queue.indexOf(callback);
     if (index === -1) {
       return;
     }
-    this.queue.splice(index, 1);
+    queue.splice(index, 1);
   }
 
   broadcast() {
-    this.queue.forEach(callback => callback());
+    const routeMatched = this.queue.reduce(
+      (acc, nextCallback) => nextCallback() || acc,
```

Comment:
1. here you call the callback to compute path matching (for route, and for "not found route" it make no sense)

Reply:
And this callback is used to print (or not) the common routes

Reply:
Reading the code, I think this call we also hide not found route.
And then you call manually "hide not found routes". (You have to, but this is what I wanted to point out, the callback as 2 meaning, and it could be hard to understand for maintenance purpose)